src/services/goals.js

```diff
@@ -466,27 +468,53 @@ export async function saveGoalsForReport(goals, report) {
     const goalId = goal.id;
     const fields = goal;
 
-    if (!Number.isInteger(goalId)) {
+    let newGoal;
+
+    if (Number.isInteger(goalId)) {
+      newGoal = await Goal.update(fields, { where: { id: goalId } });
+    } else {
       delete fields.id;
+      // In order to reuse goals with matching text we need to do the findOrCreate as the
+      // upsert would not preform the extrea checks and logic now required.
+      [newGoal] = await Goal.findOrCreate({
+        where: {
+          grantId: fields.grantId,
+          name: fields.name,
+          status: { [Op.not]: 'Closed' },
+        },
+        defaults: fields,
+      });
     }
 
-    // using upsert here and below
-    // - add returning: true to options to get an array of [<Model>,<created>] (postgres only)
-    // - https://sequelize.org/v5/class/lib/model.js~Model.html#static-method-upsert
-    const [newGoal] = await Goal.upsert(fields, { returning: true });
+    // This linkage of goal directly to a report will allow a save to be made even if no objective
+    // has been started.
+    // A hook on ActivityReportGoal will automatically update the status of the goal if required.
+    await ActivityReportGoal.findOrCreate({
```

Comment:
I don't remember seeing any place (might have just missed it!) where this linkage is undone. If I add a goal to a report and then decide I don't want that goal on my report